tests/integration/test_task_pool.py

```diff
@@ -2025,3 +2025,31 @@ async def test_remove_no_respawn(flow, scheduler, start, log_filter):
         assert (
             z1 is None
         ), '1/z should have stayed removed (but has been added back into the pool'
+
+
+async def test_set_future_tasks_in_new_flow(
+    flow, scheduler, run, complete, log_filter
+):
+    id_ = flow({
+        "scheduler": {"allow implicit tasks": True},
+        "scheduling": {
+            "graph": {"R1": "first => good & ugly?\nugly:failed? => recover"}},
+        "runtime": {
+            "ugly": {
+                "script": (
+                    "if (( CYLC_TASK_SUBMIT_NUMBER == 1 )); then"
+                    "\n    false\nfi"
+                )
+            },
+            "recover": {
+                "script": (
+                    'cylc set ${CYLC_WORKFLOW_ID} //1/good --flow=2\n'
+                    'cylc trigger ${CYLC_WORKFLOW_ID}//1/first --flow=2'
```

Comment:
Running actual bash scripts including Cylc commands makes this a functional test, not an integration test.

Reply:
I think it should be possible to test this using `start` rather than `run` by calling:

* `schd.pool.set_prereqs_and_outputs` (to set the output in flow 2)
* `schd.pool.force_trigger_tasks` or `schd.pool.get_or_spawn_task` (to spawn the upstream task in flow 2)
* `schd.pool.spawn_on_outputs` (to test the spawning of downstream tasks from upstream outputs)

Reply:
I'll give it a go.